extensions/amp-story/1.0/amp-story.js

```diff
@@ -440,13 +442,20 @@ export class AmpStory extends AMP.BaseElement {
     if (isExperimentOn(this.win, 'amp-story-branching')) {
       this.registerAction('goToPage', invocation => {
         const {args} = invocation;
-        if (args) {
-          this.storeService_.dispatch(
-            Action.SET_ADVANCEMENT_MODE,
-            AdvancementMode.GO_TO_PAGE
-          );
-          this.switchTo_(args['id'], NavigationDirection.NEXT);
+        if (!args) {
+          return;
         }
+        this.storeService_.dispatch(
+          Action.SET_ADVANCEMENT_MODE,
+          AdvancementMode.GO_TO_PAGE
+        );
+        // If open, closes the sidebar before navigating.
+        const promise = this.storeService_.get(StateProperty.SIDEBAR_STATE)
+          ? Services.historyForDoc(this.getAmpDoc()).goBack()
```

Comment:
curious: why do we do this instead of doing a `dispatch(TOGGLE_SIDEBAR, false)` to close the sidebar?

Reply:
Setting `SIDEBAR_STATE` to false would update the Story UI but would not run any code within `amp-sidebar`. We'd still have the sidebar history entry, its listeners, etc.